Bottom: d76464e95ed9ba22a6d45a79aa7e2524aaf0922c
Top:    e81baa064d30dc65b21e611f7624f774ecb71a4a
Author: Johannes Rudolph <johannes_rudolph@gmx.de>
Date:   2009-10-05 22:27:38 +0200

split Local interface in two: ROLocal[+T] with load and Local[T] extends ROLocal[T] with store


---

diff --git a/bytecode/src/main/scala/net/virtualvoid/bytecode/Bytecode.scala b/bytecode/src/main/scala/net/virtualvoid/bytecode/Bytecode.scala
index 7095b89..d4822e9 100644
--- a/bytecode/src/main/scala/net/virtualvoid/bytecode/Bytecode.scala
+++ b/bytecode/src/main/scala/net/virtualvoid/bytecode/Bytecode.scala
@@ -109,8 +109,10 @@ object Bytecode{
     											  ,thenB:F[R]=>Nothing):F[R]
   }
   
-  trait Local[T]{
-    def load[ST<:List]:F[ST] => F[ST**T]
+  trait ROLocal[+T]{
+    def load[R<:List,T2>:T]:F[R] => F[R**T2]
+  }  
+  trait Local[T] extends ROLocal[T]{
     def store[ST<:List]:F[ST**T] => F[ST]
   }
